src/components/molecules/CardPile/CardPile.tsx

```diff
@@ -0,0 +1,152 @@
+/* eslint-disable max-lines */
+import { CardPileMenu } from '@atoms/CardPileMenu';
+import { PlayingCard } from '@atoms/PlayingCard';
+import { PlayingCardFace } from '@atoms/PlayingCardFace';
+import type { CardHandle } from '@models/CardHandle';
+import type { CardProps } from '@models/CardProps';
+import { shuffleArray } from '@utils/array-utils';
+import classnames from 'classnames';
+import React, { FC, KeyboardEvent, useMemo, useRef, useState } from 'react';
+import { createUseStyles } from 'react-jss';
+import { v4 as uuid } from 'uuid';
+
+/** Describes the properties for card pile */
+export interface CardPileProps<T extends CardProps = CardProps> {
+  /** The cards metadata */
+  cards: T[];
+  /** The component function to be used for rendering */
+  component?: FC<T>;
+  /** The component function to be used for back face rendering */
+  backFace?: FC;
+}
+
+const useStyles = createUseStyles({
+  cardPile: {
+    display: 'grid',
+    gridTemplateColumns: '1fr',
+    gridTemplateRows: '1fr',
+    width: 171,
+    margin: '0 auto',
+    position: 'relative',
+    cursor: 'pointer',
+  },
+  multiItem: {
+    '&:before': {
+      width: 90,
+      height: '97%',
+      content: '""',
+      position: 'absolute',
+      background: 'repeating-linear-gradient( to right, #a2a2a2, #808080 2px, #c5c5c5 4px, #c5c5c5 0px)',
+      right: -8,
+      border: 'solid 1px black',
+      borderRadius: '12px',
+    },
+  },
+  firstItem: { zIndex: 3 },
+  secondItem: { zIndex: 2 },
+  lastItem: { zIndex: 1 },
+  card: {
+    gridArea: '1 / 1 / 1 / 1',
+  },
+  overlay: {
+    position: 'absolute',
+    width: '100%',
+    height: '100%',
+    zIndex: '10',
+    background: 'rgba(0, 0, 0, 0.5)',
+    '&:before': {
+      content: '"x"',
+      position: 'absolute',
+      width: '1.5em',
+      height: '1.5em',
+      color: 'white',
+      display: 'flex',
+      justifyContent: 'center',
+      alignItems: 'center',
+      transform: 'translate(-50%, -50%)',
+      top: '50%',
+      left: '50%',
+      fontSize: '2em',
+      border: 'solid white 0.25rem',
+      borderRadius: '50%',
+    },
+  },
+  cardPileMenu: {
+    position: 'absolute',
+    left: '-100%',
+  },
+});
+
+// eslint-disable-next-line import/prefer-default-export
+export const CardPile: FC<CardPileProps> = ({ cards, component = PlayingCard, backFace = PlayingCardFace }) => {
+  const { cardPile, multiItem, card, firstItem, secondItem, lastItem, overlay, cardPileMenu } = useStyles();
```

Comment:
I don't really like using destructuring for class names. It looks like we are using a cardPile object, a card, etc. Why don't you do `const classes = useStyles();` and then use them like `className={classes.cardPile}`?

Reply:
I don't like it either!
I had the impression it was my linter complaining, but perhaps it was on another case.
I will move it back to classes